### DISCUSSION TITLE
Tried to read to pickle file but a "ModuleNotFoundError: No module named 'mmdet3d'" error occurred

### DISCUSSION THREAD

**Discussion by BuddyZhu** [open-mmlab/mmdetection3d#2247] [General] [answered]
![Pickle failed](https://user-images.githubusercontent.com/13614743/215942401-a79ccf69-0c35-4bd6-8765-8898c933c976.jpg)
I retrained 3DSSD and tried to test the model, the tested result was saved to a pickle file, but no matter if I tried to open it with VScode extension or through python codes as the attached picture shows, same error occurred.
I am wondering why we need "mmdet3d" module to read the pickle file?
How should I solve this problem? 
Thanks in advance for any help!

**Comment by JingweiZhang12** [comment]
Maybe you can try to load it with MMEngine.

**Reply by BuddyZhu** [reply]
![2023-02-01_103607](https://user-images.githubusercontent.com/13614743/216132422-61cb4e08-f477-4a6a-8736-e2bcc61f5935.jpg)
Thanks for getting back, I tried, still not working. I am trying to install mmdet3d.

**Reply by BuddyZhu** [reply]
I installed mmengine under the environment of mmdetection3d, it worked.
Just feel weird why we need module 'mmdet3d' to open this pickle file.

**Comment by JingweiZhang12** [comment, marked as answer]
@BuddyZhu Hi, how did you save the results? I guess maybe you save the box structures `LiDARInstance3DBoxes` rather than `np.ndarray` so the pkl file parsing will import mmdet3d.
You can set `pklfile_prefix` to specify the saved path in the command script and we will save the results https://github.com/open-mmlab/mmdetection3d/blob/dev-1.x/mmdet3d/evaluation/metrics/kitti_metric.py#L438

**Reply by BuddyZhu** [reply]
I followed the following commands, and the pickle file was saved to the specified path.
`python tools/test.py configs/3dssd_ray/3dssd_4x1_kitti-3d-3class.py  work_dirs/3dssd_4x1_kitti-3d-3class/latest.pth --out ./work_dirs/3dssd_4x1_kitti-3d-3class/test_result/3dssd_4x1_kitti-3d-3class_80epoch.pkl --show --show-dir ./work_dirs/3dssd_4x1_kitti-3d-3class/test_result`
for test.py, I didn't see `pklfile_prefix` in the parameter list.

**Comment by JingweiZhang12** [comment]
Both are fine. Your solution is on the master/dev branch while my solution is on the dev-1.x branch: specificly, `--cfg-options test_evaluator.pklfile_path=${SAVED_PATH}`
